core/test/XiosRead_test.cpp

```diff
@@ -0,0 +1,145 @@
+    xios_handler.createAxis("axis_A");
+    xios_handler.setAxisSize("axis_A", n1);
+    xios_handler.setAxisValues("axis_A", { 0, 1 });
+    xios_handler.createAxis("axis_B");
+    xios_handler.setAxisSize("axis_B", n2);
+    xios_handler.setAxisValues("axis_B", { 0, 1, 2 });
+    xios_handler.createAxis("axis_C");
+    xios_handler.setAxisSize("axis_C", n3);
+    xios_handler.setAxisValues("axis_C", { 0, 1, 2, 3 });
+    xios_handler.createAxis("axis_D");
+    xios_handler.setAxisSize("axis_D", n4);
+    xios_handler.setAxisValues("axis_D", { 0, 1, 2, 3, 4 });
+
+    // Grid setup
+    xios_handler.createGrid("grid_2D");
+    xios_handler.gridAddAxis("grid_2D", "axis_A");
+    xios_handler.gridAddAxis("grid_2D", "axis_B");
+    xios_handler.createGrid("grid_3D");
+    xios_handler.gridAddAxis("grid_3D", "axis_A");
+    xios_handler.gridAddAxis("grid_3D", "axis_B");
+    xios_handler.gridAddAxis("grid_3D", "axis_C");
+    xios_handler.createGrid("grid_4D");
+    xios_handler.gridAddAxis("grid_4D", "axis_A");
+    xios_handler.gridAddAxis("grid_4D", "axis_B");
+    xios_handler.gridAddAxis("grid_4D", "axis_C");
+    xios_handler.gridAddAxis("grid_4D", "axis_D");
+
+    // Field setup
+    xios_handler.createField("field_2D");
+    xios_handler.setFieldOperation("field_2D", "instant");
+    xios_handler.setFieldGridRef("field_2D", "grid_2D");
+    xios_handler.createField("field_3D");
+    xios_handler.setFieldOperation("field_3D", "instant");
+    xios_handler.setFieldGridRef("field_3D", "grid_3D");
+    xios_handler.createField("field_4D");
+    xios_handler.setFieldOperation("field_4D", "instant");
+    xios_handler.setFieldGridRef("field_4D", "grid_4D");
+
+    // File setup
+    xios_handler.createFile("input");
+    xios_handler.setFileType("input", "one_file");
+    xios_handler.setFileOutputFreq("input", "1ts");
+    xios_handler.setFileMode("input", "read");
+    xios_handler.setFileParAccess("input", "collective");
+    xios_handler.fileAddField("input", "field_2D");
+    xios_handler.fileAddField("input", "field_3D");
+    xios_handler.fileAddField("input", "field_4D");
+
+    xios_handler.close_context_definition();
+
+    // --- Tests for file API
+    // create arrays for data to be read into
+    double* field_2D = new double[n1 * n2];
+    double* field_3D = new double[n1 * n2 * n3];
+    double* field_4D = new double[n1 * n2 * n3 * n4];
+    // Verify calendar step is starting from zero
+    REQUIRE(xios_handler.getCalendarStep() == 0);
+    // Check the input file exists
+    REQUIRE(std::filesystem::exists("input.nc"));
+    // simulate 4 iterations (timesteps)
+    for (int ts = 1; ts <= 4; ts++) {
+        // update the current timestep
+        xios_handler.updateCalendar(ts);
+        // receive data from XIOS that is read from disk
+        xios_handler.read("field_2D", field_2D, n1, n2);
+        xios_handler.read("field_3D", field_3D, n1, n2, n3);
+        xios_handler.read("field_4D", field_4D, n1, n2, n3, n4);
```

Comment:
This doesn't currently work. I'm getting the error:
```
In file "nc4_data_input.cpp", function "virtual void xios::CNc4DataInput::readFieldAttributes_(xios::CField*, bool)",  line 171 -> Field 'field_2D' has incorrect dimension 
Verify dimension of grid defined by 'grid_ref' or 'domain_ref'/'axis_ref' and dimension of grid in read file.
```

Reply:
The issue was a bug in XIOS for the case where the grid is built only from axes (not a domain). One of the developers provided a fix for this, so I just need to update the Docker build to use latest XIOS.

Reply:
This has now been done.